ephios/core/services/notifications/backends.py

```diff
@@ -25,25 +30,55 @@ def enabled_notification_backends():
         yield from (b() for b in backends)
 
 
+def send_a_notification(backend):
+    with transaction.atomic():
+        # grab an untried notification and mark it
+        if (
+            notification := Notification.objects.exclude(backends_tried__has_key=backend.slug)
+            .exclude(backends_tried__has_key=backend.slug)
+            .first()
+        ) is None:
+            return
+        notification.backends_tried[backend.slug] = False
+        notification.save()
+
+    # send it
+    if backend.can_send(notification) and backend.user_prefers_sending(notification):
+        try:
+            backend.send(notification)
+        except Exception as e:  # pylint: disable=broad-except
```

Comment:
we had some error mails for notifications where the corresponding object has been deleted in the meantime (e.g. notification for new event but the event has been deleted). Maybe we want to delete the notification immediately if a DoesNotExistExcpetion is thrown?